base/runtime_internals.jl

```diff
@@ -173,6 +173,70 @@ false
 """
 ispublic(m::Module, s::Symbol) = ccall(:jl_module_public_p, Cint, (Any, Any), m, s) != 0
 
+_function_macro_error() = (@noinline; error("@__FUNCTION__ can only be used within a function"))
+
+"""
+    @__FUNCTION__ -> Function
```

Comment:
See https://docs.julialang.org/en/v1.13-dev/manual/documentation/#man-writing-documentation

> When included, a return type should be written after the signature, separated by `::`, while a named return value should be separated by `->`, with a space on both sides.

Also see PR #57012, PR #57583.

I think simply doing it like this might be the best:

```suggestion
    @__FUNCTION__
```

Reply:
I copied the style from the `@__MODULE__` one.

If we change it, should `@__MODULE__` also be updated to remove the `-> Module`?

Reply:
Hm, it looks like it's the same for [`@__LINE__ -> Int`](https://github.com/JuliaLang/julia/blob/d82ccc1a9390a6e6b275d0852f2d86dac3406312/base/essentials.jl#L1206), [`@__FILE__ -> String`](https://github.com/JuliaLang/julia/blob/d82ccc1a9390a6e6b275d0852f2d86dac3406312/base/loading.jl#L4132), and [`@__DIR__ -> String`](https://github.com/JuliaLang/julia/blob/d82ccc1a9390a6e6b275d0852f2d86dac3406312/base/loading.jl#L4145).

Seeing those, I'm inclined to keep it as-is. Wdyt?

Reply:
Surely the `@__MODULE__`, `@__LINE__`, `@__FILE__` and `@__DIR__` were simply not yet updated for the new style.

> Seeing those, I'm inclined to keep it as-is. Ok to resolve?

No? IMO if you don't agree with the style guide, you should make a separate PR to change the style guide.

Reply:
The style guide doesn’t say anything about macro docstrings. But it seems there’s a convention to use `-> …` for the reflection macros.

Reply:
> They were just missed in the #57012 update

Our messages crossed. Thanks for confirming! Ok ill implement